### DISCUSSION TITLE
Powershell is delayed or blocked

### DISCUSSION THREAD

**Discussion by jcbsorensen** [gokcehan/lf#1344] [Q&A] [answered]
Hello :)

I am experiencing weird behavior with LF when using powershell commands. If I enter the shell and type something like 'new-item test.txt' then LF executes the command immediately (and the test.txt file pops up). However, when I make a small powershell script inside the lfrc file and execute it (fx mapping), LF will not execute it right away - at first nothing happens and when I hit q to exit LF and subsequently return to LF then the powershell script has been executed and the file suddenly exists. It seems like the powershell script execution is being blocked when triggered from a predefined mapping from lfrc. 

**Example lfrc content:**
set shell powershell
cmd test_me &mkdir test_dir
map DD test_me

If I press DD then nothing happens .But when I return to LF then the test_dir exists. This is not the case if I execute the powershell script manually in the shell within LF.

Does anyone have experience with this issue?

I know it has been stated that powershell can be problematic for LF but I just want to ensure I am not doing anything wrong.

~Jacob

**Comment by joelim-work** [comment]
This is due to a bug in `lf` where `load` is not automatically triggered after invoking a `&` command. I have submitted a PR #1345 to fix this.

For now you can use a `%` command instead, the difference is that input/output is connected to the bottom statline.

**Reply by jcbsorensen** [reply, marked as answer]
Thanks :) It helped in some cases. But sometimes % also to cause some glitches. I seem to have resolved it by using cmd shell combined with simple python scripts.

**Reply by CatsDeservePets** [reply]
Hello @jcbsorensen!
To add to this, I also experience quite a bit of slow-down when starting `powershell` in general. This can be shortened a little bit by setting `set shellopts "-nop"`. This skips loading your profile which is often stored in OneDrive.
